### PR TITLE
fix: Correctly use MySQL for phpunit-mysql and add phpunit-mariadb for  MariaDB

### PR DESCRIPTION
**Pull request by susnux**
The workflow name and description says MySQL but MariaDB was used.

Instead use MySQL but provide an optional MariaDB template.

### CODE REVIEW

File: workflow-templates/phpunit-mariadb.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+  phpunit-mariadb:
+    runs-on: ubuntu-latest
+
+    needs: changes
+    if: needs.changes.outputs.src != 'false'
+
+    strategy:
+      matrix:
+        php-versions: ['8.0', '8.1', '8.2', '8.3']
```
Comment (nickvergessen) [inline workflow-templates/phpunit-mariadb.yml:56]: This is quite a big matrix. Maybe only test with one PHP version like with sqlite, oci and postgres?

**Review by nickvergessen — COMMENTED** [review]
I agree with the fix and split. Discussed it with Julius when text had a bug that only happened on MySQL but not Maria in summer, but we didn't find the time to execute it

**Review by skjnldsv — APPROVED** [review]

File: workflow-templates/phpunit-mariadb.yml (hunk 1 above)
Comment (susnux) [reply workflow-templates/phpunit-mariadb.yml:56]: For now I changed this, but my question is: What is the recommended database to test all PHP versions?
Currently it seems to be MySQL, but maybe we should set the default to sqlite to save some CI time (as it is faster, but  you probably want to at least test your app with every PHP version once and there is only one SQLite version)

Comment (nickvergessen) [reply workflow-templates/phpunit-mariadb.yml]: It's mysql/mariadb and should be either of those or postgres. Sqlite is not typing at all, so more likely to allow buggy code/queries in general.
If we optimize that to the one db no one should use in production we are optimizing the wrong thing or based on the wrong matrix.

I would keep it on the mysql workflow for now, so people updating that one don't accidentally drop all php matrix testing 😅

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**